server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapperTests.java

```diff
@@ -465,12 +466,51 @@ public void testKnnVectorsFormat() throws IOException {
     }
 
     @Override
-    protected SyntheticSourceSupport syntheticSourceSupport() {
+    protected IngestScriptSupport ingestScriptSupport() {
         throw new AssumptionViolatedException("not supported");
     }
 
     @Override
-    protected IngestScriptSupport ingestScriptSupport() {
-        throw new AssumptionViolatedException("not supported");
+    protected SyntheticSourceSupport syntheticSourceSupport() {
+        return new DenseVectorSyntheticSourceSupport();
+    }
+
+    @Override
+    protected boolean supportsEmptyInputArray() {
+        return false;
+    }
+
+    private static class DenseVectorSyntheticSourceSupport implements SyntheticSourceSupport {
+        private final int dims = between(5, 1000);
+        private final boolean indexed = randomBoolean();
+        private final boolean indexOptionsSet = indexed && randomBoolean();
+
+        @Override
+        public SyntheticSourceExample example(int maxValues) throws IOException {
+            List<Float> value = randomList(dims, dims, ESTestCase::randomFloat);
+            System.err.println(value.size());
+            return new SyntheticSourceExample(value, value, this::mapping);
+        }
+
+        private void mapping(XContentBuilder b) throws IOException {
+            b.field("type", "dense_vector");
+            b.field("dims", dims);
+            if (indexed) {
+                b.field("index", true);
+                b.field("similarity", "l2_norm");
+                if (indexOptionsSet) {
+                    b.startObject("index_options");
+                    b.field("type", "hnsw");
+                    b.field("m", 5);
+                    b.field("ef_construction", 50);
+                    b.endObject();
+                }
+            }
+        }
+
+        @Override
+        public List<SyntheticSourceInvalidExample> invalidExample() throws IOException {
+            return List.of();
```

Comment:
It looks to me like there is nothing like `ignore_above` or `ignore_malformed` or `doc_values: false` on this field type.